src/logic/Calculate.js

```diff
@@ -0,0 +1,19 @@
+import Operate from './Operate';
+
+const Calculate = (data, name) => {
+  let { total, next } = data;
+  const { operation } = data;
+  if (name === '+/-') {
+    total *= (-1);
+    next *= (-1);
+  }
+  const newData = {
+    total: parseFloat(total),
+    next: parseFloat(next),
+    operation,
+  };
+  Operate(newData.total, newData.next, operation);
+};
```

Comment:
- Let's change `name` to `buttonName` to make it more meaningful
   ![image](https://user-images.githubusercontent.com/36057474/74854889-3b87f680-5340-11ea-8f88-9b93be97f060.png)
- Let's add the case where `=` is been pressed. This is when we want to call the `operate` function.
- Let's add the case where `AC` is been pressed. This is when `buttonName === AC`, all values should reset.
- Let's handle the case where `buttonName` is a number. The `calculate` function should be able to concatenate numbers
- Let's also handle cases where '.' is pressed. Note that:
   - when `next` and `total` is `undefined`, "0." should be returned
   - when `next` is defined, `.` should be appended to it
- We should return the calculator objects (`total`, `next` and `operation`)
   > `calculator data object` contains three properties: `total`, `next` and `operation` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/totalNextOperate.png))